### PR TITLE
refactor: use non deprecated type in eslint configuration

### PR DESCRIPTION
**Pull request by jerome-benoit**
and remove duplicated prettier configuration

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,6 +41,7 @@
     "build": "tsup",
     "publish": "clean-publish",
     "lint": "eslint --max-warnings=0",
+    "lint:fix": "eslint --max-warnings=0 --fix",
```
Comment (AriPerkkio) [inline package.json:44]:
```suggestion
```

**Review by AriPerkkio — APPROVED** [review]
Looks good, thanks! Didn't realize we have 2 Prettier configs 😄

File: .prettierrc [hunk 2]
```diff
@@ -1,6 +0,0 @@
-{
-  "endOfLine": "auto",
```
Comment (AriPerkkio) [inline .prettierrc:2]: Looks like removing this is now failing Windows CI. Maybe prettier is merging these configs?

**Review by AriPerkkio — CHANGES_REQUESTED** [review]

File: .prettierrc (hunk 2 above)
Comment (jerome-benoit) [reply .prettierrc:2]: No, that file has priority. I will re-add it and remove the other one.

**Review by AriPerkkio — APPROVED** [review]

**Pull request merged**